### PR TITLE
fix(twitch): react 18 support

### PR DESCRIPTION
**Pull request by lennartkloock**
## Proposed changes

Adds support for react 18 and adds a completely unrelated ci doc I wrote.

## Types of changes

What types of changes does your code introduce to 7TV?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [x] I have read the [CONTRIBUTING](https://github.com/SevenTV/SevenTV/blob/main/CONTRIBUTING.md) doc
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [x] Any dependent changes have been merged and published in downstream modules

## Further comments

### CODE REVIEW

File: CHANGELOG-nightly.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
-### Untitled Version
```
Comment (TroyKomodo) [inline CHANGELOG-nightly.md:1]: Undo this so we can merge more features.

File: src/common/ReactHooks.ts [hunk 2]
```diff
@@ -14,9 +14,8 @@ export function getRootVNode(): ReactExtended.ReactVNode | undefined {
 	const element = document.querySelector(REACT_ROOT_SELECTOR);
 	if (!element) return undefined;
 
-	const root = Reflect.get(element, "_reactRootContainer");
-
-	return root?._internalRoot?.current;
+	const root = Reflect.get(element, "_reactRootContainer") || Reflect.get(element, "__reactContainer$");
+	return root;
```
Comment (TroyKomodo) [inline src/common/ReactHooks.ts:18]: This changes the return is this correct?

Comment (TroyKomodo) [reply src/common/ReactHooks.ts:18]: for react 16 we returned `root._internalRoot.current` vs now we always just return `root`.

Comment (lennartkloock) [reply src/common/ReactHooks.ts:18]: Yes, idk why but this is correct

**Review by TroyKomodo — APPROVED** [review]
Works on dev testing site.

**Pull request merged**